### PR TITLE
Testing infrastructure cleanup

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs [hunk 1]
```diff
@@ -98,7 +119,7 @@ public HttpClient CreateClient(Uri baseAddress, params DelegatingHandler[] handl
             else
             {
 
-                for (var i = handlers.Length - 1; i > 1; i++)
+                for (var i = handlers.Length - 1; i > 1; i--)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs]: lol

File: src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs [hunk 2]
```diff
@@ -21,43 +17,69 @@ public class WebApplicationTestFixture<TStartup> : IDisposable where TStartup :
     {
         private readonly TestServer _server;
 
+        /// <summary>
+        /// Bootstraps an MVC application with the assemblies in the <typeparamref name="TStartup"/> assembly
+        /// dependency context and a content root that asumes that the application is on the
+        /// src\AssemblyName folder relative to the solution file.
+        /// </summary>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs]: This comment is way too dense to really digest, It needs to be broken up.
```
<para>
Creates a TestServer instance using the MVC application defined by <typeparamref name="TStartup"/>. The startup code defined in <typeparamref name="TStartup"/> will be executed to configure the application.
</para>
<para>
This constructor will infer the application root directive by searching for a solution file (*.sln) and then appending the path <c>src/{AssemblyName}</c> to the solution directory. The application root directory will be used to discover views and content files.
</para>
<para>
The application assemblies will be loaded from the dependency context of the assembly containing <typeparamref name="TStartup"/>. This means that project dependencies of the assembly containing <typeparamref name="TStartup"/> will be loaded as application assemblies.
</para>

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Infrastructure/MvcSampleFixture.cs [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+using System.IO;
+using System.Reflection;
+
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
     public class MvcSampleFixture<TStartup> : MvcTestFixture<TStartup>
         where TStartup : class
     {
-        public MvcSampleFixture() : base("samples") { }
+        public MvcSampleFixture() : base(Path.Combine("samples", typeof(TStartup).GetTypeInfo().Assembly.GetName().Name)) { }
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Infrastructure/MvcSampleFixture.cs]: You probably don't need the `GetTypeInfo()` call anymore. `Type.Assembly` is a thing now.

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Infrastructure/MvcSampleFixture.cs]: Same comment applies elsewhere.

File: src/Microsoft.AspNetCore.Mvc.Testing/MvcWebApplicationBuilder.cs [hunk 4]
```diff
@@ -24,8 +22,6 @@ namespace Microsoft.AspNetCore.Mvc.Testing
     /// <typeparam name="TStartup">The application startup class.</typeparam>
     public class MvcWebApplicationBuilder<TStartup> where TStartup : class
     {
-        private TestCulture _systemCulture;
-
         public string ContentRoot { get; set; }
         public IList<Action<IServiceCollection>> ConfigureServicesBeforeStartup { get; set; } = new List<Action<IServiceCollection>>();
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Testing/MvcWebApplicationBuilder.cs]: Would we want these to settable since you can clear them?

**Pull request merged**